registration/views/onsite_admin.py

```diff
@@ -138,25 +168,49 @@ def onsite_admin(request):
 @staff_member_required
 def onsite_admin_search(request):
     event = Event.objects.get(default=True)
-    terminals = list(Firebase.objects.all())
     query = request.POST.get("search", None)
     if query is None:
         return redirect("registration:onsite_admin")
 
-    errors = []
-    results = Badge.objects.filter(
-        Q(attendee__lastName__icontains=query)
-        | Q(attendee__preferredName__icontains=query)
-        | Q(attendee__firstName__icontains=query),
-        Q(event=event),
-    ).prefetch_related("attendee", "event")
-    if len(results) == 0:
-        errors = [
-            {"type": "warning", "text": 'No results for query "{0}"'.format(query)}
-        ]
+    data = []
 
-    context = {"terminals": terminals, "errors": errors, "results": results}
-    return render(request, "registration/onsite-admin.html", context)
+    def collectBadges(badges):
+        for badge in badges:
+            data.append({
+            "id": badge.id,
+            "edit_url": reverse("admin:registration_badge_change", args=(badge.id,)),
+            "attendee": {
+                "firstName": badge.attendee.firstName,
+                "lastName": badge.attendee.lastName,
+                "preferredName": badge.attendee.preferredName,
+            },
+            "badgeName": badge.badgeName,
+            "badgeNumber": badge.badgeNumber,
+            "abandoned": badge.abandoned,
+        })
+
+    query = query.strip()
+
+    try:
+        badge_id = int(query)
+        badges = Badge.objects.filter(event=event, badgeNumber=badge_id)
+        collectBadges(badges)
+    except:
+        pass
+
+    fullName = Func(F("attendee__firstName"), Value(" "), F("attendee__lastName"), function="CONCAT")
+    greaterSimilarity = Func("name_similarity", "badge_similarity", function="GREATEST")
+
+    results = Badge.objects.annotate(
+        name_similarity=TrigramSimilarity(fullName, query),
```

Comment:
As I understand from the docs, the trigram extension is not provided by default in Postgres.  We'll want to [add a migration](https://docs.djangoproject.com/en/4.2/ref/contrib/postgres/operations/#django.contrib.postgres.operations.TrigramExtension) for this PR to ensure the `pg_trgm` gets installed.

Reply:
Ah whoops, I guess I must've manually created it at some point, will add that migration